auth/genericoauth/provider.go

```diff
@@ -141,9 +142,12 @@ func (GenericTeamProvider) ProviderConstructor(
 	}
 
 	var oauthVerifier verifier.Verifier
-	if genericOAuth.Scope != "" {
+	if genericOAuth.Scope != "" && genericOAuth.Authority = "" {
 		oauthVerifier = NewScopeVerifier(genericOAuth.Scope)
-	} else {
+	} else if genericOAuth.Scope = "" && genericOAuth.Authority != "" {
+		oauthVerifier = NewAuthorityVerifier(genericOAuth.Authority)
+	}
+	else {
```

Comment:
Will take a look. I added Go support to my Atom editor and it should run automatically already.